src/commands/dev/lib/dev-server.ts

```diff
@@ -281,8 +284,28 @@ export default class DevServer {
     // then re-run the build that generated this asset
     if (this.shouldRebuild(req) && asset.buildEntry) {
       const entrypoint = relative(this.cwd, asset.buildEntry.fsPath);
-      this.output.debug(`Rebuilding asset: ${entrypoint}`);
-      await executeBuild(nowJson, this, asset);
+      const buildTimestamp: number = asset.buildTimestamp || 0;
+      let buildPromise = this.inProgressBuilds.get(entrypoint);
+      if (buildPromise) {
```

Comment:
Maybe we can move this `if (buildPromise) { return false }` into `this.shouldRebuild`? I feel this could made logic here looks simpler.

Reply:
I'll refactor it soon, but I want to get another canary release out now. I like your suggestion 👍